mpc_controller_node:
  ros__parameters:
    # General controller parameters
    velocity_estimation_topic: /vcu_msgs/velocity_estimation
    steering_topic: /vcu_msgs/steering_feedback
    reference_topic: /planning/reference
    car_command_topic: /control/car_command
    controller_frequency: 40.0
    controller_timeout_reset: 1000
    controller_timeout_fatal: 10000

    ax_command_standstill: 5.0
    vel_threshold_standstill: 5.0

    max_radius_path_points: 25.0

    # MPC controller parameters
    solver_type: EMBOTECH # ACADOS or EMBOTECH
    consecutive_failures_limit: 200

    # visualization means publishing messages that allow to visualize the MPC controller e.g. in RViz
    enable_visualisation: true

    # logging means logging the MPC controller data by sending information via messages
    enable_logging: true

    # profiling with easy_profiler
    enable_profiling: true

    # MPC output interpolation
    enable_output_interpolation: true

    # MPC delay compensation (predicting the initial state at the time of finishing the optimization)
    enable_delay_compensation: false

    # Velocity planning means that the MPC controller will take the velocity profile into account
    enable_velocity_planning: true

    # Plan from car or Project onto path
    mpc_plan_from_car: false
    mpc_from_car_pruning_distance: 2.0

    # additional time for interpolation after the predicted initial state in seconds
    mpc_dt_interp_ax: 0.010
    mpc_dt_interp_dels: 0.010
    mpc_dt_interp_dpsi: 0.010

    # time constants for the first order behavior assumed for steering and acceleration
    tau_ax: 0.01
    tau_dels: 0.035

    # MPC Model parameters
    m: 200.0
    l_f: 0.884
    l_r: 0.646
    C_d: 1.6
    C_r: 0.15
    C_l: 5.3
    car_width: 1.5
    B_tire: 11.0
    C_tire: 1.3
    D_tire: 1.2
    h_cg: 0.27
    Iz: 240.0
    l_to_front: 1.8
    l_to_rear: 1.3

    # MPC Cost parameters
    q_ds: 0.08
    q_ds_terminal: 0.0
    q_n: 0.1
    q_n_terminal: 12.0
    q_mu: 2.0
    q_mu_terminal: 5.0
    r_dax: 0.0001
    r_ddels: 0.001

    # MPC Constraint parameters
    vx_max: 20.0
    vx_max_terminal: 3.0
    axt_abs_max: 6.0
    ayt_abs_max: 7.0
    dels_abs_max: 0.46
    dax_min: -150.0
    dax_max: 150.0
    ddels_abs_max: 5.0

    terminal_vel_progress_scaling: 1.0
    terminal_vel_scaling: 1.25
    trackbound_shrinkage: 0.03

mpc_visualisation_node:
  ros__parameters:
    prediction_horizon_topic: /control/prediction_horizon
    fitted_path_topic : /control/fitted_path
    ref_project_topic : /control/ref_projection
    mpc_logging_topic: /control/mpc_logging
